src/parsers/chrono/chrono.js

```diff
@@ -56,6 +56,31 @@ dayPeriodsParser.extract = (text, ref, match) => {
   });
 };
 
+const currentMonthDayParser = Object.assign(
+  new chrono.Parser(),
+  {
+    pattern: () => /\b(?:on|by)\s+the\s+(\d\d?)(?:th|st|rd|nd)\b/,
+    extract: (text, ref, match) => {
+      const day = Number(match[1]);
+      const currentDay = ref.getDate(); // TODO support timezones
+      const currentMonth = ref.getMonth();
+
+      const month = day >= currentDay ? currentMonth : currentMonth + 1;
+
+      return new chrono.ParsedResult({
+        ref,
+        text: match[0],
+        index: match.index,
+        start: {
+          month: month + 1,
```

Comment:
javascript dates have their months between 0 and 11. While chrono expects a month between 1 and 12. Maybe I should have this as a comment, what do you think ?